### PR TITLE
Feature/398

### PR DESCRIPTION
**Pull request by 14JAW**
Tabs did not have the functionality we were looking for. Instead of the sliding bar under the buttons, I have them highlighting. Also, the Project Name is in the center of the screen and is a shortcut to get to project settings.

---
This change is [Reviewable](https://reviewable.io/reviews/sillsdev/thecombine/438)

### CODE REVIEW

File: src/components/AppBar/AppBarComponent.tsx [hunk 1]
```diff
@@ -8,6 +8,7 @@ import Logo from "./Logo";
 import theme from "../../types/theme";
 import NavigationButtons from "./NavigationButtons";
 import { getProjectId } from "../../backend/localStorage";
+import ProjectNameButton from "./ProjectNameButton";
```
Comment (languagewarrior) [inline src/components/AppBar/AppBarComponent.tsx:11]:
```suggestion
import ProjectNameButton from "./ProjectNameButton";
```

File: src/components/AppBar/AppBarComponent.tsx [hunk 2]
```diff
@@ -25,10 +26,13 @@ function AppBarComponent() {
               spacing={2}
               alignItems="center"
             >
-              <Grid item>
+              <Grid item xs>
```
Comment (languagewarrior) [inline src/components/AppBar/AppBarComponent.tsx:29]:
```suggestion
              <Grid item xs>
```

File: src/components/AppBar/AppBarComponent.tsx [hunk 3]
```diff
@@ -25,10 +26,13 @@ function AppBarComponent() {
...
                 <Logo />
                 {getProjectId() !== "" && <NavigationButtons />}
               </Grid>
+              <Grid item xs={6}>
```
Comment (languagewarrior) [inline src/components/AppBar/AppBarComponent.tsx:33]: <Grid item xs={6}>

File: src/components/AppBar/AppBarComponent.tsx [hunk 4]
```diff
@@ -25,10 +26,13 @@ function AppBarComponent() {
...
+                <ProjectNameButton />
```
Comment (languagewarrior) [inline src/components/AppBar/AppBarComponent.tsx:34]: src/components/AppBar/AppBarComponent.tsx

File: src/components/AppBar/NavigationButtons.tsx [hunk 5]
```diff
@@ -1,25 +1,36 @@
-import React from "react";
+import React, { useState } from "react";
 import { Button } from "@material-ui/core";
 import history from "../../history";
 import { Translate } from "react-localize-redux";
 
 /** A button that redirects to the home page */
 export default function NavigationButtons() {
+  const highlight: string = "#1976d2";
```
Comment (languagewarrior) [inline src/components/AppBar/NavigationButtons.tsx:8]: const highlight: string = "#1976d2";

**Comment by charjhall** [comment]
___
*[src/components/AppBar/NavigationButtons.tsx, line 1 at r2](https://reviewable.io/reviews/sillsdev/thecombine/438#-MA6dHRu1yAFKZirly6j:-MA6dHRu1yAFKZirly6k:b-hf4vmz) ([raw file](https://github.com/sillsdev/thecombine/blob/ec02a943f945c6e418935e8464e069d155b7b9af/src/components/AppBar/NavigationButtons.tsx#L1)):*
> ```
>  useState
> ```
Unused

**Comment by charjhall** [comment]
___
*[src/components/AppBar/ProjectNameButton.tsx, line 37 at r2](https://reviewable.io/reviews/sillsdev/thecombine/438#-MA6dksP63zXXlFAV6vS:-MA6dksP63zXXlFAV6vT:b1xnn9u) ([raw file](https://github.com/sillsdev/thecombine/blob/ec02a943f945c6e418935e8464e069d155b7b9af/src/components/AppBar/ProjectNameButton.tsx#L37)):*
> ```
> history.push("/project-settings");
> ```
Since this functionality already exists here, you may want to remove the Project Settings from the user menu in `UserMenu.tsx` line 65

**Comment by charjhall** [comment]
___
*[src/components/AppBar/ProjectNameButton.tsx, line 11 at r2](https://reviewable.io/reviews/sillsdev/thecombine/438#-MA6jHPE2qE4vpfL3qll:-MA6jHPFA0fZ0Ubou9wu:bevceuv) ([raw file](https://github.com/sillsdev/thecombine/blob/ec02a943f945c6e418935e8464e069d155b7b9af/src/components/AppBar/ProjectNameButton.tsx#L11)):*
<details><summary><i alt="summary">Quoted 6 lines of code…</i></summary>

> ```
> function tabColor(currentTab: CurrentTab, tabName: CurrentTab) {
>   const colors = ["inherit", shade];
>   if (currentTab === tabName) {
>     return colors[1];
>   } else return colors[0];
> }
> ```
</details>

Not exactly sure how types works, but you could define this method in the `currentTab.tsx` file and import it, since you use it here and in `NavigationButtons.tsx`

**Comment by charjhall** [comment]
___
*[src/components/AppBar/ProjectNameButton.tsx, line 10 at r2](https://reviewable.io/reviews/sillsdev/thecombine/438#-MA6l-koEPCVTkV89BMv:-MA6l-koEPCVTkV89BMw:bevceuv) ([raw file](https://github.com/sillsdev/thecombine/blob/ec02a943f945c6e418935e8464e069d155b7b9af/src/components/AppBar/ProjectNameButton.tsx#L10)):*
<details><summary><i alt="summary">Quoted 6 lines of code…</i></summary>

> ```
> function tabColor(currentTab: CurrentTab, tabName: CurrentTab) {
>   const colors = ["inherit", shade];
>   if (currentTab === tabName) {
>     return colors[1];
>   } else return colors[0];
> }
> ```
</details>

Not exactly sure how types works, but you could define this method in the `currentTab.tsx` file and import it, since you use it here and in `NavigationButtons.tsx`

**Comment by charjhall** [comment]
___
*[src/components/AppBar/ProjectNameButton.tsx, line 27 at r2](https://reviewable.io/reviews/sillsdev/thecombine/438#-MA6o91aAIqlowS9v2Zf:-MA6o91aAIqlowS9v2Zg:bcktfxe) ([raw file](https://github.com/sillsdev/thecombine/blob/ec02a943f945c6e418935e8464e069d155b7b9af/src/components/AppBar/ProjectNameButton.tsx#L27)):*
> ```
> getProjectId()
> ```
I think this should be refactored to retrieve the ID from the store via redux. This will require refactoring `AppBarComponent.tsx` to receive props from the store via redux

**Comment by charjhall** [comment]
___
*[src/components/AppBar/NavigationButtons.tsx, line 8 at r2](https://reviewable.io/reviews/sillsdev/thecombine/438#-MA6p1ayB5cKq0wrWMjt:-MA6p1ayB5cKq0wrWMju:b-m5a342) ([raw file](https://github.com/sillsdev/thecombine/blob/ec02a943f945c6e418935e8464e069d155b7b9af/src/components/AppBar/NavigationButtons.tsx#L8)):*
<details><summary><i alt="summary">Quoted 6 lines of code…</i></summary>

> ```
> function tabColor(currentTab: CurrentTab, tabName: CurrentTab) {
>   const colors = ["inherit", shade];
>   if (currentTab === tabName) {
>     return colors[1];
>   } else return colors[0];
> }
> ```
</details>

See comment on `ProjectNameButton.tsx`

**Comment by charjhall** [comment]
___
*[src/components/AppBar/NavigationButtons.tsx, line 1 at r2](https://reviewable.io/reviews/sillsdev/thecombine/438#-MA6pHYZ3zdJoMYrD-QJ:-MA6pHYZ3zdJoMYrD-QK:brkhweo) ([raw file](https://github.com/sillsdev/thecombine/blob/ec02a943f945c6e418935e8464e069d155b7b9af/src/components/AppBar/NavigationButtons.tsx#L1)):*
> ```
>  useState
> ```
Unused import

**Comment by charjhall** [comment]
___
*[src/components/AppBar/AppBarComponent.tsx, line 44 at r2](https://reviewable.io/reviews/sillsdev/thecombine/438#-MA6pYDb-RorofA6Ju3o:-MA6pYDb-RorofA6Ju3p:bl83p9s) ([raw file](https://github.com/sillsdev/thecombine/blob/ec02a943f945c6e418935e8464e069d155b7b9af/src/components/AppBar/AppBarComponent.tsx#L44)):*
> ```
> getProjectId()
> ```
See comment on `ProjectNameButton.tsx`

**Comment by charjhall** [comment]
___
*[src/components/AppBar/ProjectNameButton.tsx, line 27 at r2](https://reviewable.io/reviews/sillsdev/thecombine/438#-MA6o91aAIqlowS9v2Zf:-MA6zWfX7ZYLxvN2iC4j:b-dfb4po) ([raw file](https://github.com/sillsdev/thecombine/blob/ec02a943f945c6e418935e8464e069d155b7b9af/src/components/AppBar/ProjectNameButton.tsx#L27)):*
<details><summary><i>Previously, charjhall (Charles Hall) wrote…</i></summary><blockquote>

> ```
> getProjectId()
> ```
I think this should be refactored to retrieve the ID from the store via redux. This will require refactoring `AppBarComponent.tsx` to receive props from the store via redux
</blockquote></details>

This will be addressed in a later issue

**Comment by charjhall** [comment]
___
*[src/components/AppBar/tests/AppBarComponent.test.tsx, line 11 at r2](https://reviewable.io/reviews/sillsdev/thecombine/438#-MA7-bMI6PZlL7bQ9CZj:-MA7-bMJDC-0RV3GnxUN:b30cqb4) ([raw file](https://github.com/sillsdev/thecombine/blob/ec02a943f945c6e418935e8464e069d155b7b9af/src/components/AppBar/tests/AppBarComponent.test.tsx#L11)):*
> ```XML
> const createMockStore = configureMockStore([]);
> 
> it("renders without crashing", () => {
> ```

You may want to add a test to ensure only one of the tabs is selected

**Comment by charjhall** [comment]
___
*[src/components/ProjectSettings/ProjectSettingsComponent.tsx, line 98 at r2](https://reviewable.io/reviews/sillsdev/thecombine/438#-MA73TjNB3Ru27DR9sfI:-MA73TjNB3Ru27DR9sfJ:b-34cl7k) ([raw file](https://github.com/sillsdev/thecombine/blob/ec02a943f945c6e418935e8464e069d155b7b9af/src/components/ProjectSettings/ProjectSettingsComponent.tsx#L98)):*
> ```
> <AppBarComponent currentTab={CurrentTab.ProjectSettings} />
> ```
This should update on selecting a different project. The way that SeHee implemented the selection is to change the value in the store. If you use the state in `AppBarCompnent.tsx` that should fix this issue. For now, I think we are leaving it until we refactor `AppBarComponent.tsx` to receive props from the store

**Comment by charjhall** [comment]
___
*[src/components/AppBar/tests/AppBarComponent.test.tsx, line 31 at r5](https://reviewable.io/reviews/sillsdev/thecombine/438#-MASlymQ1diPGxGbdcqV:-MASlymQ1diPGxGbdcqW:b-dkor3s) ([raw file](https://github.com/sillsdev/thecombine/blob/00e0b3319e3fd1edacc53ad6c75ebdbc18dd1f2c/src/components/AppBar/tests/AppBarComponent.test.tsx#L31)):*
> ```XML
>       .create(
>         <Provider store={mockStore}>
>           <NavigationButtons currentTab={CurrentTab.DataCleanup} />
> ```

This ensures that only one of the Navigation Buttons is shaded,  but does not ensure that the Project Button is not also shaded

**Pull request merged**